boxsdk/object/search.py

```diff
@@ -5,12 +5,147 @@
 from boxsdk.config import API
 from .base_endpoint import BaseEndpoint
 from boxsdk.util.translator import Translator
+import json
+
+class MetadataSearchFilter(object):
+    """
+    Helper class to encapsulate a single search filter. A search filter can only search against one template,
+    but can filter on many fields.
+    See :class:`MetadataSearchFilters`.
+    """
+    def __init__(self, template_key, scope):
+        """
+        :param template_key:
+            The key of the template to search on
+        :type template_key:
+            `unicode`
+        :param scope:
+            The scope of the template to search on
+        :type scope:
+            `unicode`
+        """
+        self._template_key = template_key
+        self._scope = scope
+        self._field_filters = {}
+
+    def as_dict(self):
+        """
+        Returns a `dict` representation of this object
+
+        :return:
+            The `dict` representation
+        :rtype:
+            `dict`
+        """
+        return {
+            'templateKey': self._template_key,
+            'scope': self._scope,
+            'filters': self._field_filters
+        }
+
+    def add_value_based_filter(self, field_key, value):
+        """
+        Add a value-based filter (used for token-based search on string fields, and exact match search on all other fields)
+
+        :param field_key:
+            The field key to filter on
+        :type field_filter:
+            `unicode`
+        :param value:
+            The value to use to filter
+        :type value:
+            `unicode`
+        """
+        self._field_filters.update({field_key: value})
+
+    def add_range_filter(self, field_key, gt_value = None, lt_value = None):
+        """
+        Add a range filter (used for ranged searches on numbers and dates)
+
+        :param field_key:
+            The field key to filter on
+        :type field_filter:
+            `unicode`
+        :param gt_value:
+            The lower bound of the range filter (inclusive)
+        :type gt_value:
+            `unicode` or `int` or `float` or `long` or None
+        :param lt_value:
+            The upper bound of the range filter (inclusive)
+        :type lt_value:
+            `unicode` or `int` or `float` or `long` or None
+        """
+        range_part = {}
+        if gt_value:
+            range_part['gt'] = gt_value
+        if lt_value:
+            range_part['lt'] = lt_value
+        self._field_filters.update({field_key: range_part})
+
+
+class MetadataSearchFilters(object):
```

Comment:
I would suggest import MetadataSearchFilters in boxsdk/_ _ init _ _.py, since it would be used in the search API in client.